build.gradle

```diff
@@ -83,3 +76,6 @@ processResources
     }
 }
 
+jar {
+    from configurations.shade.collect { it.isDirectory() ? it : zipTree(it) }
+}
```

Comment:
Alternatively, you might want to do:
```java
jar {
    configurations.shade.each {
        from(it.isDirectory() ? it : zipTree(it)) { exclude 'META-INF', 'META-INF/**' }
    }
}
```

That way you don't include any of the `META-INF` stuff. I don't _think_ it matters, but it might be safer.